config/routes.rb

```diff
@@ -4,8 +4,10 @@
   # Defines the root path route ("/")
   # root "articles#index"
   root "users#index"
-
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      post "new_comment", to: "comments#create", on: :member
+      post "like_post", to: "likes#create", on: :member
```

Comment:
- [ ] For the `like` and `comment`, I would suggest the use of restricted routes. This is the same as you have done with `users` and `posts`. It would read:

```ruby
resources :comments, only: [:new, :create]
resources :likes, only: [:new, :create]
```